blocks/monitoring/aggregation.py

```diff
@@ -138,11 +138,62 @@ def get_aggregator(self):
         return aggregator
 
 
-def mean(numerator, denominator=1.):
+def mean(numerator, denominator=1., name=None):
     """Mean of quantity (numerator) over a number (denominator) values."""
     variable = numerator / denominator
     variable.tag.aggregation_scheme = Mean(numerator, denominator)
-    variable.name = numerator.name
+    if name is not None:
```

Comment:
Yes, but matrix, shared etc do add them.

The variable after aggregation will typically be named - because it will be used for reports and such. I added the fix because I first used named_copy to set the name, which removed the aggregation scheme. Adding the name like that is IMHO helpful.